adapters/resetdigital/resetdigital.go

```diff
@@ -4,305 +4,237 @@ import (
-	if !found {
-		return nil, []error{fmt.Errorf("no matching impression found for ImpID %s", resetDigitalBid.ImpID)}
+	if len(bidResp.SeatBid) == 0 {
+		return nil, nil
 	}
 
-	bid, err := getBidFromResponse(resetDigitalBid)
-	if err != nil {
-		return nil, []error{err}
+	return parseBidResponse(request, &bidResp)
+}
+
+func parseBidResponse(request *openrtb2.BidRequest, bidResp *openrtb2.BidResponse) (*adapters.BidderResponse, []error) {
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+	var errs []error
+
+	if bidResp.Cur != "" {
+		bidResponse.Currency = bidResp.Cur
+	} else {
+		bidResponse.Currency = currencyUSD
 	}
 
-	bidType := GetMediaTypeForImp(requestImp)
+	for _, seatBid := range bidResp.SeatBid {
+		for i := range seatBid.Bid {
+			if seatBid.Bid[i].Price <= 0 {
+				errs = append(errs, &errortypes.Warning{
+					Message: fmt.Sprintf("price %f <= 0 filtered out", seatBid.Bid[i].Price),
+				})
+				continue
+			}
 
-	bidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
-	bidResponse.Currency = "USD" // Default currency
-	bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
-		Bid:     bid,
-		BidType: bidType,
-		Seat:    openrtb_ext.BidderName(resetDigitalBid.Seat),
-	})
+			bidType, err := getBidType(seatBid.Bid[i], request)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
 
-	return bidResponse, nil
-}
+			seat := openrtb_ext.BidderName(bidderSeat)
+			if seatBid.Seat != "" {
+				seat = openrtb_ext.BidderName(seatBid.Seat)
+			}
 
-// findRequestImpByID searches for an impression by its ID in the list of impressions
-func findRequestImpByID(imps []openrtb2.Imp, impID string) (openrtb2.Imp, bool) {
-	for _, imp := range imps {
-		if imp.ID == impID {
-			return imp, true
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &seatBid.Bid[i],
+				BidType: bidType,
+				Seat:    seat,
+			})
 		}
 	}
-	return openrtb2.Imp{}, false
-}
-
-func getBidFromResponse(bidResponse *resetDigitalBid) (*openrtb2.Bid, error) {
 
-	bid := &openrtb2.Bid{
-		ID:    bidResponse.BidID,
-		Price: bidResponse.CPM,
-		ImpID: bidResponse.ImpID,
-		CID:   bidResponse.CID,
-		CrID:  bidResponse.CrID,
-		AdM:   bidResponse.HTML,
-	}
+	return bidResponse, errs
+}
 
-	w, err := strconv.ParseInt(bidResponse.W, 10, 64)
-	if err != nil {
-		return nil, err
+func getBidType(bid openrtb2.Bid, request *openrtb2.BidRequest) (openrtb_ext.BidType, error) {
+	if bid.MType > 0 {
+		switch bid.MType {
+		case openrtb2.MarkupBanner:
+			return openrtb_ext.BidTypeBanner, nil
+		case openrtb2.MarkupVideo:
+			return openrtb_ext.BidTypeVideo, nil
+		case openrtb2.MarkupAudio:
+			return openrtb_ext.BidTypeAudio, nil
+		case openrtb2.MarkupNative:
+			return openrtb_ext.BidTypeNative, nil
+		}
 	}
-	bid.W = w
 
-	h, err := strconv.ParseInt(bidResponse.H, 10, 64)
-	if err != nil {
-		return nil, err
+	if len(request.Imp) == 1 {
+		if request.Imp[0].ID != bid.ImpID {
```

Comment:
Just double-checking: we don't care about `bid.ImpID` if `bid.MType`s value is between 1 and 4 as written into the `switch` statement above, correct?